src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs

```diff
@@ -829,12 +829,22 @@ private void LaunchStreams(ConnectionState connectionState, Dictionary<string, o
             {
                 _ = _sendIAsyncStreamItemsMethod
                     .MakeGenericMethod(reader.GetType().GetInterface("IAsyncEnumerable`1")!.GetGenericArguments())
-                    .Invoke(this, new object[] { connectionState, kvp.Key.ToString(), reader, cts });
+                    .Invoke(this, [connectionState, kvp.Key.ToString(), reader, cts]);
                 continue;
             }
-            _ = _sendStreamItemsMethod
-                .MakeGenericMethod(reader.GetType().GetGenericArguments())
-                .Invoke(this, new object[] { connectionState, kvp.Key.ToString(), reader, cts });
+            else
+            {
+                if (ReflectionHelper.TryGetStreamType(reader.GetType(), out var channelGenericType))
+                {
+                    _ = _sendStreamItemsMethod
+                        .MakeGenericMethod(channelGenericType)
+                        .Invoke(this, [connectionState, kvp.Key.ToString(), reader, cts]);
+                    continue;
```

Comment:
Nit: let's use `if ... continue; if ... continue ...` or `if ... else if ... else`. Mixing the `continue` and `if/else` blocks is way harder to follow.